packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.controller.js

```diff
@@ -6,47 +8,184 @@ export default class {
+        if (existingCapability) {
+          if (!existingCapability.sourceEngines.includes(curr.sourceEngine)) {
+            existingCapability.sourceEngines.push(curr.sourceEngine);
+          }
+          if (
+            !existingCapability.destinationEngines.includes(
+              curr.destinationEngine,
+            )
+          ) {
+            existingCapability.destinationEngines.push(curr.destinationEngine);
+          }
+        } else {
+          const capabilityArray = {
+            type: curr.type,
+            parameters: curr.parameters,
+            sourceEngines: [curr.sourceEngine],
+            destinationEngines: [curr.destinationEngine],
+          };
+          acc.push(capabilityArray);
+        }
+        return acc;
+      },
+      [],
+    );
   }
 
   cancel() {
-    this.trackDashboard(`service_integration::add_${this.engineName}_cancel`);
+    this.trackDashboard(
+      `service_integration::add_${this.database.engine}_cancel`,
+    );
     this.goBack();
   }
 
+  handleIntegrationChange() {
+    const { sourceEngines, destinationEngines } = this.integrationCapability;
+    // Filter services list according to integration capability
+    if (sourceEngines === destinationEngines) {
+      const filteredServices = this.servicesList.filter((db) =>
+        sourceEngines.includes(db.engine),
+      );
+      this.sourceServiceList = filteredServices;
+      this.destinationServiceList = filteredServices;
+    } else {
+      this.sourceServiceList =
+        isEqual(sourceEngines, [this.database.engine]) &&
+        !destinationEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) => sourceEngines.includes(db.engine));
+      this.destinationServiceList =
+        isEqual(destinationEngines, [this.database.engine]) &&
+        !sourceEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) =>
+              destinationEngines.includes(db.engine),
+            );
+    }
+    // Preselect service if only one element
+    this.sourceService =
+      this.sourceServiceList.length === 1 ? this.sourceServiceList[0] : null;
+    this.destinationService =
+      this.destinationServiceList.length === 1
+        ? this.destinationServiceList[0]
+        : null;
+    // Emtpy parameters array
+    this.parameters = {};
+    // Disable select inputs if needed
+    const isServiceSourceListEmpty = this.sourceServiceList.length === 0;
+    const isCurrentEngineInSourceList = sourceEngines.includes(
+      this.database.engine,
+    );
+    const isServiceDestinationListEmpty =
+      this.destinationServiceList.length === 0;
+    const isCurrentEngineInDestinationList = destinationEngines.includes(
+      this.database.engine,
+    );
+    // Disable select input if needed
+    this.isDisabledSourceSelect =
+      isServiceSourceListEmpty ||
+      (isCurrentEngineInSourceList && !isCurrentEngineInDestinationList);
```

Comment:
To reduce the number of lines in this method, this could be extracted to a getter as follow
```javascript
get isDisabledSourceSelect() {
  if (!this.sourceServiceList || this.sourceServiceList.length === 0) {
    return true;
  }
  return
    this.integrationCapability.sourceEngines.includes(this.database.engine) &&
    !this.integrationCapability.destinationEngines.includes(this.database.engine))
}

Reply:
Works like a charm, thanks ! https://github.com/ovh/manager/commit/ab8e0199587826e2f4dd2ce1f47bbe4086b79303